napari/_vispy/overlays/scale_bar.py

```diff
@@ -120,10 +120,19 @@ def _on_data_change(self):
                 # set scale color negative of theme background.
                 # the reason for using the `as_hex` here is to avoid
                 # `UserWarning` which is emitted when RGB values are above 1
-                background_color = get_theme(
-                    self.viewer.theme, False
-                ).canvas.as_hex()
-                background_color = transform_color(background_color)[0]
+                if (
+                    hasattr(self.viewer, 'window')
+                    and hasattr(self.viewer.window, "_qt_viewer")
+                    and self.viewer.window._qt_viewer.canvas.background_color_override
```

Comment:
Not great that we have to use private state here (or more generally that the scale bar has access to the whole of the `viewer`), but I can't think of anything better right now.

Reply:
I have fixed this by use the parent of node and allowing to use of this fix with `ViewerModel` not only `Viewer`.